client/components/data/purchases/site-purchases/index.jsx

```diff
@@ -1,65 +0,0 @@
-/**
- * External dependencies
- */
-import React from 'react';
-
-/**
- * Internal dependencies
- */
-import { fetchSitePurchases } from 'lib/upgrades/actions';
-import PurchasesStore from 'lib/purchases/store';
-import StoreConnection from 'components/data/store-connection';
-import sitesFactory from 'lib/sites-list';
-
-/**
- * Module variables
- */
-const sites = sitesFactory(),
-	stores = [
-		PurchasesStore,
-		sites
```

Comment:
Ah, sounds good. Looks like we were rendering twice when the site changed before then, because of the handler in the data component.